android-kotlin/app/src/main/java/com/mongodb/app/ItemAdapter.kt

```diff
@@ -32,11 +35,24 @@ internal class ItemAdapter(data: OrderedRealmCollection<Item>, private val confi
         val obj: Item? = getItem(position)
         holder.id = obj?._id
         holder.name.text = obj?.name
+        holder.checkbox.isChecked = obj?.checked?:false
+
+        /**
+         *  Allows a user to check and uncheck an item and updates its status in the realm.
```

Comment:
I would prefer line comment syntax (not javadoc) within methods.

Reply:
extracted the methods and moved the comments